fms/testing/fms_hf_equivalence.py

```diff
@@ -0,0 +1,314 @@
+import argparse
+import json
+import logging
+import os
+import sys
+from collections import OrderedDict
+from dataclasses import dataclass, field
+from typing import List
+
+import torch
+from transformers import AutoModelForCausalLM, AutoTokenizer
+
+from fms.models import get_model
+from fms.models.hf.utils import to_hf_api
+
+
+_device = "cuda"
+_decice_for_diff = "cpu"
+
+dtype_mapping = {"float16": torch.float16, "float32": torch.float32}
+
+batch_size = 8
+max_length = 2048
+
+
+def generate_and_write_tokens(
+    model, tokenizer, input_json_path, output_dir, model_impl_type
+):
+    """
+    This function loads the prompts, generate the tokens for the given prompts using the supplied model and tokenizer.
+    Finally, writes the generated tokens in a file "output_dir/model_impl_type/tokens" The output files contains serialized
+    ordered dict that contains two entries for each prompt, one for the input tokens and one for the generated tokens.
+    Paddings are removed before writing the tokens so it's safe to compare the outputs of two different runs with different batch sizes.
+
+     Parameters
+    ----------
+    model: Either of the HF or HF adapted FMS model
+    tokenizer: The tokenizer associated with the model
+    input_json_path: File path to the prompt file. The prompt file is expected to contain a set of JSON records with one attribute - "prompt".
+    output_dir: Path to the output directory
+    model_impl_type: One of the "fms" or "hf"
+    """
+    prompts = load_prompts(input_json_path)
```

Comment:
do we also want to accept standard datasets?

Reply:
Sure, makes sense. Can do that in the next iteration in a separate PR, once we decide if this utility going to be useful